cli/flox-core/src/activations.rs

```diff
@@ -10,31 +10,105 @@ use serde::{Deserialize, Serialize};
 use serde_json::json;
 use time::OffsetDateTime;
 
-use crate::{path_hash, Version};
+use crate::path_hash;
 
 type Error = anyhow::Error;
 
+/// Latest supported version for compatibility between:
+/// - `flox` and `flox-activation-scripts`
+/// - `flox-activations` and `flox-watchdog`
+///
+/// Incrementing this will require existing activations to exit.
+const LATEST_VERSION: u8 = 1;
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct UncheckedVersion(u8);
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CheckedVersion(u8);
```

Comment:
@ysndr pushed some changes that I need to finish incorporating into the tests. A review from a third-person, both on the PR direction and those changes, would still be useful in the meantime.